atomic_reactor/inner.py

```diff
@@ -430,18 +429,18 @@ def __init__(
         client_version: str = None,
     ):
         """
+        :param build_dir: a directory holding all the artifacts to build an image.
+        :type build_dir: RootBuildDir
         :param source: where/how to get source code to put in image
         :param plugins: the plugins to be executed in this workflow
         :param user_params: user (and other) params that control various aspects of the build
         :param reactor_config_path: path to atomic-reactor configuration file
         :param plugin_files: load plugins also from these files
         :param client_version: osbs-client version used to render build json
         """
-        self._build_dir = build_dir
+        self.build_dir = build_dir
```

Comment:
I would still call it `build_dirs`, I think that more closely matches how we want to use this object (to handle multiple build dirs)

Reply:
Yeah, that's a good point. I'm just afraid `workflow.build_dir.for_each()` is going to look weird now. Maybe the methods could be renamed to `for_each_platform` and `for_all_platforms_copy`? But maybe the current naming is fine, not sure

Reply:
Alternatively, maybe plugins shouldn't access `build_dirs.path`. Maybe there should be predefined directories for everything that doesn't go in the platform-specific build dirs - like `build_dirs.source_container_inputs`, `build_dirs.output`

Reply:
> Yeah, that's a good point. I'm just afraid workflow.build_dir.for_each() is going to look weird now. Maybe the methods could be renamed to `for_each_platform` and `for_all_platforms_copy`? But maybe the current naming is fine, not sure

I like these two new names. 😃 I think the new name reflects the intention much straightforward than the old ones.